### PR TITLE
Use 'Create File From Template' KW in 'Compile Inference Service YAML'

### PR DESCRIPTION
**Pull request by manosnoam**
Replace the SED commands in 'Compile Inference Service YAML', 
with the Common keyword 'Create File From Template'.

This removes the multiple call to external SED commands (i.e. multiple processes), 
by invoking a single keyword to replace variables in the template: caikit_isvc.yaml

### CODE REVIEW

**Comment by bdattoma** [comment]
lgtm. Could we hold on merging this, though? It's an improvement we can have after the release.

**Comment by manosnoam** [comment]
> lgtm. Could we hold on merging this, though? It's an improvement we can have after the release.

@bdattoma I guess we merge now,  1.32 release is out

**Comment by bdattoma** [comment]
> > lgtm. Could we hold on merging this, though? It's an improvement we can have after the release.
> 
> @bdattoma I guess we can merge now, 1.32 release is out

Could you please fix the conflicts and re-test? Because more test cases were added after this PR.

File: ods_ci/tests/Resources/Files/llm/caikit_isvc.yaml [hunk 1]
```diff
@@ -5,17 +5,17 @@ metadata:
...
-    minReplicas: {{MIN_REPLICAS}}
-    scaleTarget: {{SCALE_TARGET}}
-    canaryTrafficPercent: {{CanaryTrafficPercent}}
-    minReplicas: {{MIN_REPLICAS}}
-    serviceAccountName: {{SA_NAME}}
+    scaleMetric: ${scaleMetric}
+    minReplicas: ${min_replicas}
+    scaleTarget: ${scaleTarget}
+    canaryTrafficPercent: ${canaryTrafficPercent}
+    minReplicas: ${min_replicas}
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Files/llm/caikit_isvc.yaml:15]: could you remove this line? it's duplicated

File: ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/422__model_serving_llm.robot [hunk 2]
```diff
@@ -661,36 +661,17 @@ Compile Inference Service YAML
...
-          ...    sed -i 's/{{SCALE_METRIC}}/${scaleMetric}/g' ${LLM_RESOURCES_DIRPATH}/caikit_isvc_filled.yaml
-    END
-    IF   '${canaryTrafficPercent}' == '${EMPTY}'
-        ${rc}    ${out}=    Run And Return Rc And Output
-        ...    sed -i '/canaryTrafficPercent/d' ${LLM_RESOURCES_DIRPATH}/caikit_isvc_filled.yaml
-    ELSE
-        ${rc}    ${out}=    Run And Return Rc And Output
-        ...    sed -i 's/{{CanaryTrafficPercent}}/${canaryTrafficPercent}/g' ${LLM_RESOURCES_DIRPATH}/caikit_isvc_filled.yaml
+        ${scaleTarget}=    Set Variable    ${EMPTY}
+        ${scaleMetric}=    Set Variable    ${EMPTY}
```
Comment (bdattoma) [inline ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/422__model_serving_llm.robot:666]: could you delete the lines from yaml when autoscaling is none? like done for https://github.com/red-hat-data-services/ods-ci/pull/900/files#diff-343c068d5fc035598a61515b186f393cdae65b3e09fbc9e4326d4bbc6f03da33R675

Comment (bdattoma) [reply ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/422__model_serving_llm.robot:666]: we can check this in a next PR, don't want to block this PR more time.

**Review by bdattoma — DISMISSED** [review]

**Comment by manosnoam** [comment]
Verified after rebase. 
The Yaml created for example on test `Verify User Can Set Requests And Limits For A Model`:
rhods-smoke/3403/artifact/ods-ci/ods_ci/tests/Resources/Files/llm/caikit_isvc_filled.yaml:
image

**Comment by bdattoma** [comment]
> Verified after rebase. The Yaml created for example on test `Verify User Can Set Requests And Limits For A Model`: rhods-smoke/3403/artifact/ods-ci/ods_ci/tests/Resources/Files/llm/caikit_isvc_filled.yaml: image



it doesn't look right, because the requests and limits are missing

File: ods_ci/tests/Resources/Files/llm/caikit_isvc.yaml (hunk 1 above)
Comment (manosnoam) [reply ods_ci/tests/Resources/Files/llm/caikit_isvc.yaml:15]: Yeah, `minReplicas: {{MIN_REPLICAS}}` was there twice in the original file, I'll remove it.

**Review by lugi0 — APPROVED** [review]
Need to remove the duplicated `minReplicas` field in the yaml, other than that lgtm

**Review by bdattoma — APPROVED** [review]

**Review by lugi0 — APPROVED** [review]

**Pull request merged**